language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/builder/ResourcePathCompletionUtil.java

```diff
@@ -211,19 +242,50 @@ private static ResourceAccessPathPart getResourceAccessPartForSegment(PathSegmen
         String paramType = FunctionCompletionItemBuilder
                 .getFunctionParameterSyntax(pathParameterSymbol, context).orElse("");
         String computedSignature = "[" + paramType + "]";
-        String computedInsertText = "[${" + placeHolderIndex + ":" + defaultValue.orElse("") + "}]";
+        String computedInsertText =
+                typeSymbol.typeKind().equals(TypeDescKind.SINGLETON) ? "[" + defaultValue.orElse("") + "]" :
+                        "[${" + placeHolderIndex + ":" + defaultValue.orElse("") + "}]";
         ResourceAccessPathPart resourceAccessPathPart =
                 new ResourceAccessPathPart(computedInsertText, computedSignature);
+
+        //  A resource method parameter can be a singleton or a union with a `string`. As the node "text" is always
+        //  resolved to `string`, we need to check if typeSymbol is either a supertype or a subtype  of `string`.
         if (context.currentSemanticModel().isPresent() &&
-                context.currentSemanticModel().get().types().STRING.subtypeOf(typeSymbol)) {
-            resourceAccessPathPart.namedPathSignature = "<path>";
-            resourceAccessPathPart.namedPathInsertText = "${" + placeHolderIndex + ":" + "path" + "}";
-            resourceAccessPathPart.computedPathInsertText = "[${" + placeHolderIndex + ":" + "\"path\"" + "}]";
-            resourceAccessPathPart.isStringPathPram = true;
+                isStringSubtype(typeSymbol, context.currentSemanticModel().get().types().STRING)) {
```

Comment:
Do we need to pass the 2nd argument?

Reply:
The `context` is only visible within the function body and not across the entire class; that's why I had to pass the `String` type symbol as a parameter. If we can access the `String` type symbol statically, then we do not need to pass this explicitly.